File: ts/import-content-model/import_model_create_snippet.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = client.addContentTypeSnippet()
  .withData(builder => {
    return {
      name: 'Metadata',
      codename: 'metadata',
      elements: [
        builder.textElement({
          name: 'Title',
          codename: 'title',
          type: 'text'
        }),
        builder.textElement({
          name: 'Keywords',
          codename: 'keywords',
          type: 'text'
        }),
        builder.textElement({
          name: 'Description',
          codename: 'description',
          type: 'text'
        })
      ]
    };
  })
  .toPromise();

File: net/import-content-model/import_model_create_snippet.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = await client.CreateContentTypeSnippetAsync(new ContentTypeSnippetCreateModel
{
    Name = "Metadata",
    Codename = "metadata",
    Elements = new ElementMetadataBase[]
    {
        new TextElementMetadataModel
        {
            Name = "Title",
            Codename = "title",
        },
        new TextElementMetadataModel
        {
            Name = "Keywords",
            Codename = "keywords",
        },
        new TextElementMetadataModel
        {
            Name = "Description",
            Codename = "description",
        },
    }
});
